lib/ansible/modules/cloud/smartos/vmadm.py

```diff
@@ -0,0 +1,609 @@
+        return None
+
+    # Lookup table for the state to be in, and which command to use for that.
+    # vm_state: [vmadm commandm, forceable?]
+    cmds = {
+        'stopped': ['stop', True],
+        'running': ['start', False],
+        'deleted': ['delete', True],
+        'rebooted': ['reboot', False]
+    }
+
+    if p['force'] and cmds[vm_state][1]:
+        force = '-F'
+    else:
+        force = ''
+
+    cmd = 'vmadm {0} {1} {2}'.format(cmds[vm_state][0], force, vm_uuid)
+
+    (rc, stdout, stderr) = module.run_command(cmd)
+
+    match = re.match('^Successfully.*', stderr)
+    if match:
+        return True
+    else:
+        return False
+
+
+def create_payload(module, uuid):
+    """Create the JSON payload (vmdef) and return the filename."""
+
+    p = module.params
+
+    # Filter out the few options that are not valid VM properties.
+    module_options = ['debug', 'force', 'state']
+    vmattrs = filter(lambda prop: prop not in module_options, p)
+
+    vmdef = {}
+
+    for attr in vmattrs:
+        if p[attr]:
+            vmdef[attr] = p[attr]
+
+    try:
+        vmdef_json = json.dumps(vmdef)
+    except Exception as e:
+        module.fail_json(msg='Could not create valid JSON payload: {0}'.format(e))
+
+    # Create the temporary file that contains our payload, and set tight
+    # permissions for it may container sensitive information.
+    try:
+        # XXX: When there's a way to get the current ansible temporary directory
+        # drop the mkstemp call and rely on ANSIBLE_KEEP_REMOTE_FILES to retain
+        # the payload (thus removing the `save_payload` option).
+        fname = tempfile.mkstemp(uuid)[1]
+        fh = open(fname, 'w')
+        os.chmod(fname, 0o400)
+        fh.write(vmdef_json)
+        fh.close()
+    except Exception as e:
+        module.fail_json(msg='Could not save JSON payload as {0}: {1}'.format(fname, e))
+
+    return fname
+
+
+def vm_state_transition(module, uuid, vm_state):
+    ret = set_vm_state(module, uuid, vm_state)
+
+    # Whether the VM changed state.
+    if ret is None:
+        return False
+    elif ret:
+        return True
+    else:
+        module.fail_json(msg='Failed to set VM {0} to state {1}'.format(uuid, vm_state))
+
+
+def is_valid_uuid(uuid):
+    if re.match('^[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}$', uuid, re.IGNORECASE):
+        return True
+    else:
+        return False
+
+
+def validate_uuids(module):
+    # Perform basic UUID validation.
+    failed = []
+
+    for u in [['uuid', module.params['uuid']],
+              ['image_uuid', module.params['image_uuid']]]:
+        if u[1] and u[1] != '*':
+            if not is_valid_uuid(u[1]):
+                failed.append(u[0])
+
+    if len(failed) > 0:
+        module.fail_json(msg='No valid UUID(s) found for: {0}'.format(", ".join(failed)))
+
+
+def manage_all_vms(module, vm_state, changed):
+    """ Handle operations for all VMs, which can by definition only
```

Comment:
I would either use everywhere the simple comments ('# ...`) or use everywhere the comment blocks (`""" ... """`) but I wouldn't combine both.